copier/tools.py

```diff
@@ -147,7 +140,8 @@ def handle_remove_readonly(
 ) -> None:
     """Handle errors when trying to remove read-only files through `shutil.rmtree`.
 
-    This handler makes sure the given file is writable, then re-execute the given removal function.
+    On Windows, `shutil.rmtree` does not handle read-only files very well. This handler
+    makes sure the given file is writable, then re-execute the given removal function.
```

Comment:
It seems that`shutil.rmtree` doesn't only have this issue under Python <3.8 on Windows. See the Windows CI checks when I had this handler removed: https://github.com/copier-org/copier/actions/runs/5573531416

It seems okay to remove the custom `TemporaryDirectory` class though. I hope that's actually correct and not just uncovered by tests.

Reply:
Is the `handle_remove_readonly` utility still used somewhere else in the code?

Reply:
No, only in [`template.Template._cleanup`](https://github.com/copier-org/copier/blob/a27366bc72927919914fbe09987ffd16518f0ad8/copier/template.py#L213).